neural_compressor/torch/algorithms/weight_only/rtn.py

```diff
@@ -19,174 +19,209 @@
 # limitations under the License.
 
 
+from collections import OrderedDict
+
 import torch
 
+from neural_compressor.common.utils import RTN
+from neural_compressor.torch.algorithms import Quantizer, algo_quantizer_register
 from neural_compressor.torch.utils import get_device, logger, set_module
-from neural_compressor.torch.utils.auto_accelerator import auto_detect_accelerator
 
 from .utility import quant_tensor, search_clip
 
 
-@torch.no_grad()
-def rtn_quantize(
-    model,
-    dtype="int",
-    bits=4,
-    scheme="sym",
-    group_size=32,
-    group_dim=1,
-    quantile=1.0,
-    weight_config={},
-    export_compressed_model=False,
-    use_full_range=False,
-    use_mse_search=False,
-    **kwargs,
-):
-    """Quant the model with round to nearest method and inplace is True.
+@algo_quantizer_register(name=RTN)
+class RTNQuantizer(Quantizer):
+    def __init__(self, config: OrderedDict = {}):
+        super().__init__(config)
+        self._preprocess_config()
```

Comment:
@yiliu30 I assume that this change aims to allow user recover the quantization with config_mapping recovered from json file, or I don't know how user can get the config_mapping that INC designed.

Reply:
preprocess config step was moved back to algorithm_entry.py